aws/transit_gateway_test.go

```diff
@@ -0,0 +1,294 @@
+package aws
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/aws/aws-sdk-go/aws"
+	awsgo "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/gruntwork-io/cloud-nuke/util"
+	"github.com/gruntwork-io/gruntwork-cli/errors"
+)
+
+func createTestTransitGateway(t *testing.T, session *session.Session, name string) ec2.TransitGateway {
+	svc := ec2.New(session)
+
+	tgwName := ec2.TagSpecification{
+		ResourceType: awsgo.String(ec2.ResourceTypeTransitGateway),
+		Tags: []*ec2.Tag{
+			{
+				Key:   aws.String("Name"),
+				Value: aws.String(name),
+			},
+		},
+	}
+
+	param := &ec2.CreateTransitGatewayInput{
+		TagSpecifications: []*ec2.TagSpecification{&tgwName},
+	}
+
+	result, err := svc.CreateTransitGateway(param)
+	require.NoError(t, err)
+	require.True(t, len(aws.StringValue(result.TransitGateway.TransitGatewayId)) > 0, "Could not create test TransitGateway")
+
+	//TransitGateway takes some time to be available and there isn't Waiters available yet
+	//To avoid test errors, I'm introducing a sleep call
+	time.Sleep(180 * time.Second)
+	tgw := *result.TransitGateway
+	return tgw
```

Comment:
```suggestion
      return *result.TransitGateway
```

Reply:
Implemented! Could you check again, please?